### ISSUE TITLE
App is getting crash on click on the push notification in some devices 

### ISSUE THREAD

**Issue by Monika-Paliwal** [firebase/quickstart-android#1257]
Android device: Samsung A30
Android OS version: 10
Firebase : firebase-core:18.0.0
**Steps to reproduce:**
One user call to the another user and the app is in background mode when push is coming on click on push app is getting crash.

**Observed Results:**
though, the notification does appear in the system tray, the data available when we put the data in intent in service but when we are getting the data in onCreate, the extras is null.

**Expected Results:**
when App State is Background and a Notification with data is sent, the notification appears in the system and the data will be available in the extras of the intent.
If we are using setFullScreenIntent then app is getting crash because in some devices the full screen is not opened. 
 Intent fullScreenIntent = new Intent(this, VideoCallActivity.class);
        fullScreenIntent.putExtra("NAME", NAME);
        fullScreenIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK
                | Intent.FLAG_ACTIVITY_CLEAR_TOP
                | Intent.FLAG_ACTIVITY_SINGLE_TOP
                | Intent.FLAG_ACTIVITY_CLEAR_TASK);

        PendingIntent fullScreenPendingIntent = PendingIntent.getActivity(this, 0,
                fullScreenIntent, PendingIntent.FLAG_UPDATE_CURRENT);

        String channelName = getString(R.string.call_service);
        String NOTIFICATION_CHANNEL_ID = getString(R.string.call_notification_channel_id);
        NotificationChannel chan = new NotificationChannel(NOTIFICATION_CHANNEL_ID,
                channelName, NotificationManager.IMPORTANCE_HIGH);
        chan.setLightColor(Color.BLUE);
        chan.setLockscreenVisibility(Notification.VISIBILITY_PUBLIC);
        manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        assert manager != null;
        manager.createNotificationChannel(chan);

        //set intents and pending intents to call service on click of "dismiss" action button of notification
        Intent acceptIntent = new Intent(this, IncomingCallService.class);
        acceptIntent.setAction(ACTION_ACCEPT);
        PendingIntent piAccept = PendingIntent.getService(this, 0, acceptIntent, 0);

        //set intents and pending intents to call service on click of "snooze" action button of notification
        Intent endIntent = new Intent(this, IncomingCallService.class);
        endIntent.setAction(ACTION_END);
        PendingIntent piEnd = PendingIntent.getService(this, 0, endIntent, 0);

        Notification notification = new NotificationCompat.Builder(this,
                NOTIFICATION_CHANNEL_ID)
                .setSmallIcon(R.drawable.call_incoming)
                .setContentTitle("incoming call")
                .setContentText("Name")
                .setPriority(NotificationCompat.PRIORITY_MAX)
                .setCategory(NotificationCompat.CATEGORY_CALL)
                .setFullScreenIntent(fullScreenPendingIntent, true)
                .addAction(R.drawable.white_phone_off_icon,
                        getString(R.string.decline), piEnd)
                .addAction(R.drawable.ic_icon_phone_white,
                        getString(R.string.accept), piAccept)
                .build();

       startForeground(CALL_NOTIFICATION_ID, notification);

**Comment by samtstern** [comment]
@Monika-Paliwal is this related to the quickstart apps in this repository or your own real app in production?  If it's your real app this seems like we should maybe transfer the issue to this repo: https://github.com/firebase/firebase-android-sdk

Also can you provide the stack trace for this crash?  Right now it's unclear if the crash comes from Firebase code or your code.

**Comment by Monika-Paliwal** [comment]
Yes this is related to firebase messaging android SDK  so we can move it.
